contracts/ReversibleICO.sol

```diff
@@ -224,6 +224,8 @@ contract ReversibleICO is IERC777Recipient {
     isNotInitialized
     {
 
+        require (_commitPhaseStartBlock > getCurrentBlockNumber(),"start block is set in the past");
```

Comment:
This is good to have for production deployments that start correctly.

Yet enforcing this stops deployments on actual blockchains with the contract in different states like deployment after commit phase, or in "stage 5" / etc.
Still that can be handled using a mock over the contract, so it should be fine for production.